src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs

```diff
@@ -353,6 +353,12 @@ internal override void Validate()
         {
             base.Validate();
 
+            //ADFS does not require client id to be in the form of a GUID.
+            if (Config.AuthorityInfo?.AuthorityType != AuthorityType.Adfs && !Guid.TryParse(Config.ClientId, out _))
+            {
+                throw new MsalClientException(MsalError.ClientIdMustBeAGuid, MsalErrorMessage.ClientIdMustBeAGuid);
+            }

```

Comment:
Overall this PR looks good. :+1: 

This implementation removes the GUID check from `ConfidentialClientApplication`, but keeps it for `PublicClientApplication`. Is there a specific reason to keep it, or is it just us wanting to make minimal change in this PR (which is understandable, too)? Basically, curious to know the answers to [@jmprieur 's earlier questions](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2852#issuecomment-907421378).

Reply:
I think there's no way to configure an App Id Uri for public client. That said, today you can use the same client id for both PCA and CCA. 

@trwalke - maybe we should remove this validation altoghter?

Reply:
only web APIs can have an App Id URI. Public clients only have the client_id. This check is still valid for public clients.

Reply:
Yes but you can still use the app Id Uri with for public clients and the portal does not do a good job of explaining why you should not do this. Using an app Id Uri will result in a malformed request exception from AAD so we might as well fail faster with this validate method for public client apps.

Reply:
Yea, it's not super clear because they do show the App ID URI in the App Registration Overview blade. But when you click it, it takes you to the Expose an API blade, and there they specify, in a tooltip, that it's only for a web API.
![image](https://user-images.githubusercontent.com/34331512/132045935-6641d967-2230-4b14-a530-426ab2901b46.png)
![image](https://user-images.githubusercontent.com/34331512/132046086-fc059614-9e4b-456e-a027-89fe62736bfc.png)

Reply:
@rayluo There are no issues with that I have come across using a non guid for the client ID. All of our tests are passing and the caching operations are working well. Also, I got confirmation from each of the SDKs in which we cache compatibility to confirm that the app id uri works in their SDK today.